src/tx/builder/index.ts

```diff
@@ -0,0 +1,569 @@
+        return 'Value must contains only object\'s like \'{key: "account_pubkey", id: "ak_lkamsflkalsdalksdlasdlasdlamd"}\''
+      }
+      if (value.length > 32) {
+        return `Expected 32 pointers or less, got ${value.length} instead`
+      }
+  }
+}
+
+function transformParams (
+  params: TxParamsCommon,
+  schema: TxField[], { denomination }: {denomination?: AeAmountFormats} = {}): any {
+  params = schema
+    .filter(([, t]) => t === FIELD_TYPES.amount)
+    .reduce(
+      (params: TxParamsCommon, [key]) => ({
+        ...params,
+        [key]: formatAmount(
+          params[key as keyof TxParamsCommon], { denomination })
+      }),
+      params
+    )
+  const schemaKeys = schema.map(([k]) => k)
+  return Object
+    .entries(params)
+    .reduce(
+      (acc: any, [key, value]: [key: string,
+        value: {
+          type: string
+          value: number
+        }]) => {
+        if (schemaKeys.includes(key)) acc[key] = value
+        if (['oracleTtl', 'queryTtl', 'responseTtl'].includes(key)) {
+          acc[`${key}Type`] = value.type === ORACLE_TTL_TYPES.delta ? 0 : 1
+          acc[`${key}Value`] = value.value
+        }
+        return acc
+      },
+      {}
+    )
+}
+
+// INTERFACE
+
+function getOracleRelativeTtl (params: any, txType: TxType): number {
+  const ttlKeys: {
+    [key in TxType]?: string
+  } = {
+    [TX_TYPE.oracleRegister]: 'oracleTtl',
+    [TX_TYPE.oracleExtend]: 'oracleTtl',
+    [TX_TYPE.oracleQuery]: 'queryTtl',
+    [TX_TYPE.oracleResponse]: 'responseTtl'
+  }
+  const ttlKey = ttlKeys[txType]
+  if (ttlKey == null) return 1
+  else if (params[`${ttlKey}Value`] > 0) return params[`${ttlKey}Value`]
+  else return params[ttlKey].value
+}
+
+/**
+ * Calculate min fee
+ * @alias module:@aeternity/aepp-sdk/es/tx/builder/index
+ * @param txType - Transaction type
+ * @param options - Options object
+ * @param options.gasLimit
+ * @param options.params - Tx params
+ * @example calculateMinFee('spendTx', { gasLimit, params })
+ */
+export function calculateMinFee (txType: TxType, { params, vsn }: {
+  gasLimit?: string | number
+  params?: Object
+  vsn?: number
+}): string {
+  const multiplier = new BigNumber(1e9) // 10^9 GAS_PRICE
+  if (params == null) {
+    return new BigNumber(DEFAULT_FEE).times(multiplier).toString(10)
+  }
+
+  let actualFee = buildFee(txType, { params: { ...params, fee: 0 }, multiplier, vsn })
+  let expected = new BigNumber(0)
+
+  while (!actualFee.eq(expected)) {
+    actualFee = buildFee(txType, {
+      params: { ...params, fee: actualFee }, multiplier, vsn
+    })
+    expected = actualFee
+  }
+  return expected.toString(10)
+}
+
+/**
+ * Calculate fee based on tx type and params
+ * @param txType
+ * @param options.params
+ * @param options.multiplier
+ * @param options.vsn
+ * @return {BigNumber}
+ */
+function buildFee (txType: TxType, { params, multiplier, vsn }:
+{params: any, multiplier: BigNumber, vsn?: number}): BigNumber {
```

Comment:
```suggestion
function buildFee (
  txType: TxType,
  { params, multiplier, vsn }: { params: any, multiplier: BigNumber, vsn?: number }
): BigNumber {
```
messing at the same line function name, parameters and return type looks difficult to read